### PR TITLE
Split whitelist and host-whitelist separated in two env vars

### PR DESCRIPTION
**Pull request by StudioEtrange**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [X] I have read the [contributing](https://github.com/linuxserver/docker-transmission/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:

With PR #141  was introduced a typo bug when whitelist are involved
And it appears that we need to control whitelist and host-whitelist separatly to cover all needed control on whitelist types (ip list AND dns named)

## Benefits of this PR and context:

Have 2 variable WHITELIST and HOST_WHITELIST to have independent control on whitelit and host-whitelist

This PR also fix #148

### CODE REVIEW

**Comment by JustasStankunas** [comment]
for some reason i am getting strange beharvior with this. if i set `whitelist` environment then it changes following parameters in settings.json (which i do not understand..)
```
    "rpc-host-whitelist": "<...>",       # this contains value that i provided with with env variable, as expected
    "rpc-host-whitelist-enabled": true,  # this is changed from false to true, as expected
    "rpc-whitelist": "127.0.0.1,::1",    # this is still present, with prior merge it should be gone?
    "rpc-whitelist-enabled": true,       # this is changed from false to true, i expected that it would not change
```
and when i am setting this in Kubernetes it fails me because even though `rpc-host-whitelist` is set, but `rpc-whitelist` deny my access..

**Comment by StudioEtrange** [comment]
What kind of value do you use for whitelist.
Its prime goal was to set a list of ip

**Comment by JustasStankunas** [comment]
well i want to run this client in my internal network without exposint to public www. so i want to run without user/pass. but if i run without user/pass - transmission insists to set rpc-host-whitelist (https://github.com/transmission/transmission/wiki/Editing-Configuration-Files). so i am setting my internal DNS name via this variable.

**Comment by StudioEtrange** [comment]
You are using the last version 3.0.0r0 ?
This one do not get this last PR which is needed for whitelist to work.
Without this PR, which seems to be your situation, the value of rpc-whitelist is wrong

Your value
rpc-whitelist": "127.0.0.1,::1",

Should be
rpc-whitelist : 192.168.*.* 
(Or something like this)

And to have this to work that you need
* WHITELIST=192.168.*.*
* This PR fix #146

**Comment by JustasStankunas** [comment]
i am using `3.00-r0-ls75`.
well the issue i am facing is, as fas as i understand, that i need to set `rpc-host-whitelist` to whatever host name or ip transmission will be reachable on (in my case let say it will be `torrent.home.local`, or `transmission.namespace.svc.cluster.local` in case of kubernetes internal communications) and set `rpc-host-whitelist-enabled` to `true` to enable this.
then i need to leave `rpc-whitelist-enabled` set to `false` (as i don't want to limit this from transmission) or set `rpc-whitelist` to my internal network range (and kubernetes internal network in case of this example)
for me it sound like this needs to be two different variables or have different variable which would compeltely disable `/config/settings.json` modification and would allow to do it "manually" and not from container init phase.

**Comment by StudioEtrange** [comment]
I understand the need about splitting setting for the two variables
I can do a PR like this

Did you try manually using transmission with those values :
* rpc-whitelist-enabled : false (this one take care of ip)
* rpc-host-whitelist-enabled : true (this one take care of DNS name)
I never managed to make it work.

**Comment by JustasStankunas** [comment]
i have not tried that, i wanted to test it yesterday, but this image automatically overwrites these values, so i will try that today or tomorrow and report back here.

**Comment by StudioEtrange** [comment]
The need to have two separates env var seems still needed in anyway I think to have more control on these settings
I have pushed a commit in this PR to do that


"split support for rpc-whitelist and rpc-host-whitelist in separated env var WHITELIST and HOST_WHITELIST"

File: root/etc/cont-init.d/20-config [hunk 1]
```diff
@@ -19,17 +19,22 @@ else
...
+	sed -i '/rpc-whitelist-enabled/c\    "rpc-whitelist-enabled": true,' /config/settings.json
+	sed -i "/\"rpc-whitelist\"/c\    \"rpc-whitelist\": \"$WHITELIST\"," /config/settings.json
+else
+	sed -i '/rpc-whitelist-enabled/c\    "rpc-whitelist-enabled": false,' /config/settings.json
+	sed -i "/\"rpc-whitelist\"/c\    \"rpc-whitelist\": \"$WHITELIST\"," /config/settings.json
+fi
+
+if [ ! -z "$HOST_WHITELIST" ]; then
 	sed -i '/rpc-host-whitelist-enabled/c\    "rpc-host-whitelist-enabled": true,' /config/settings.json
 	sed -i "/\"rpc-host-whitelist\"/c\    \"rpc-host-whitelist\": \"$WHITELIST\"," /config/settings.json
```
Comment (JustasStankunas) [inline root/etc/cont-init.d/20-config:31]: `$WHITELIST` needs to be changed to `$HOST_WHITELIST`

File: root/etc/cont-init.d/20-config [hunk 2]
```diff
@@ -19,17 +19,22 @@ else
...
-	sed -i '/rpc-whitelist-enabled/c\    "rpc-host-whitelist-enabled": true,' /config/settings.json
-	sed -i "/\"rpc-whitelist\"/c\    \"rpc-host-whitelist\": \"$WHITELIST\"," /config/settings.json
 else
 	sed -i '/rpc-host-whitelist-enabled/c\    "rpc-host-whitelist-enabled": false,' /config/settings.json
 	sed -i "/\"rpc-host-whitelist\"/c\    \"rpc-host-whitelist\": \"$WHITELIST\"," /config/settings.json
```
Comment (JustasStankunas) [inline root/etc/cont-init.d/20-config:34]: `$WHITELIST` needs to be changed to `$HOST_WHITELIST`

File: root/etc/cont-init.d/20-config (hunk 1 above)
Comment (JustasStankunas) [inline root/etc/cont-init.d/20-config:31]: `$WHITELIST` needs to be changed to `$HOST_WHITELIST`

**Review by JustasStankunas — CHANGES_REQUESTED** [review]
while new if for `HOST_WHITELIST` id added, but within that if `WHITELIST` is still being used.

File: root/etc/cont-init.d/20-config (hunk 2 above)
Comment (StudioEtrange) [reply root/etc/cont-init.d/20-config:34]: Yes, my bad.

File: root/etc/cont-init.d/20-config (hunk 1 above)
Comment (StudioEtrange) [reply root/etc/cont-init.d/20-config:31]: Yes, my bad.

**Comment by Noxalus** [comment]
This is a good change! 👍 
Do you know when it will be merged?

**Comment by JustasStankunas** [comment]
I have broke my test lab and can't verify this at the moment, so if @Noxalus  you could verify and confirm that this actually works. i think @aptalca could approve this merge. :)

**Comment by Noxalus** [comment]
I will do this now! I plan to build the docker image locally from [this repo](https://github.com/StudioEtrange/docker-transmission), can you confirm me that you only want me to try the RPC/RPC-Host fix and not to launch a test suite (that I didn't find 🤔 ).

**Comment by Noxalus** [comment]
I've just tested the `WHITELIST` environment variable with a fresh Docker image built with the repo linked in my previous message, and it works as expected.

The fields `rpc-whitelist` and `rpc-whitelist-enabled` are updated and `rpc-host-whitelist` and `rpc-host-whitelist-enabled` stay unchanged! 👍

**Comment by StudioEtrange** [comment]
ping ?

**Comment by aptalca** [comment]
Is this ready?

**Comment by m4rkireland** [comment]
Looking forward to this getting merged

**Comment by StudioEtrange** [comment]
> Is this ready and tested?

IMHO i have finished the PR and tested it

**Comment by wayward-turnip** [comment]
Thank you for fixing this

**Pull request merged**